src/main/java/com/project/store/controller/category/CategoryController.java

```diff
@@ -40,17 +41,15 @@ public List<CategoryDto> getAllCategories() {
         return categoryMapper.fromEntitiesToDtos(categoryList);
     }
 
-    @GetMapping("/{id}/products")
-    public CategoryWithProductsDto getCategoryWithProducts(@PathVariable Integer id) {
-        Category category = categoryService.getCategory(id);
-
-        return categoryMapper.fromEntityToDtoWithProducts(category);
-    }
-
-    @GetMapping("/categories-with-products")
-    public List<CategoryWithProductsDto> getAllCategoriesWithProducts() {
-        List<Category> categoryList = categoryService.getAllCategories();
-        return categoryMapper.fromEntitiesToDtosWithProducts(categoryList);
+    @GetMapping(value = {"/products", "/{id}/products"})
```

Comment:
You can use a non-required @RequestParam instead.

Reply:
Added @RequestParam for getting the two results